3p/ampcontext.js

```diff
@@ -24,7 +24,14 @@ export class AmpContext {
    *  @param {Window} win The window that the instance is built inside.
    */
   constructor(win) {
-    this.setupMetadata_();
+    this.win_ = win;
+
+    // if the window sentinel is set directly, that means we are running
+    // in a situation where we couldn't attach the name attribute before
+    // creating the iframe, so don't try to parse the metadata
+    if (!this.getWindowSentinel_()) {
```

Comment:
so went back to keith about this, apparently he misunderstood what I meant when I first asked him if this is explicitly only for safeframe. there are potentially other use cases, where what is assigned to the window variable will either be only a sentinel, or the context object. so I changed the name of the window variable, and changed how the function handles it